### PR TITLE
[#118] Improve generated test code

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms karsten.thoms@itemis.de

### CODE REVIEW

**Comment by LorenzoBettini** [comment]
Please have a look at my comment/suggestion on the issue https://github.com/eclipse/xtext-core/issues/118

**Comment by kthoms** [comment]
I still disagree. Right, the test tests that the DSL model is parsed. The test never fails, even if the model is invalid. The error is not detected. This leads to confusion.

See for example:
http://stackoverflow.com/questions/12703318/testing-that-something-fails-to-parse-in-xtext
https://www.eclipse.org/forums/index.php?t=rview&goto=1742997#msg_1742997

**Comment by cdietrich** [comment]
then at least we need a result.eResource.error.isempty

**Comment by spoenemann** [comment]
> then at least we need a result.eResource.error.isempty

Yes, this solution could make it more clear that we're only interested in parser errors, not validation errors.

**Comment by kthoms** [comment]
OK for me, I'll update the PR.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/junit/Junit4Fragment2.xtend [hunk 1]
```diff
@@ -82,19 +82,18 @@ class Junit4Fragment2 extends AbstractStubGeneratingFragment {
...
 			@�runWith�(�xtextRunner�)
 			@�injectWith�(�injectorProvider�)
-			class �exampleRuntimeTest�{
-			
+			class �exampleRuntimeTest� {
 				@�Inject�
 				�parseHelper�<�rootType�> parseHelper
-			
-				@�test� 
+
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/junit/Junit4Fragment2.xtend]: @kthoms is this tab here by intention?

Comment (kthoms) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/junit/Junit4Fragment2.xtend]: Yes, it is. It is basically a newline between the injection and the test method. The tab is there in this new line with class-level indentation.

**Comment by svenefftinge** [comment]
+1 for the change. 
Generally I prefer to generate less code, especially generate-once stuff.
The kind of stuff we do here should be something for a wizard not for the generator.
We just put it here, because we didn't want to have compile errors after the wizard did his job.

**Comment by cdietrich** [comment]
@kthoms is this pull request uptodate?

**Comment by kthoms** [comment]
Yes, looks so.

**Comment by cdietrich** [comment]
thx @kthoms

**Comment by dhuebner** [comment]
@cdietrich Could you please verify that core build failure is not caused by the merged change?http://services.typefox.io/open-source/jenkins/job/xtext-core/job/master/220/

**Comment by cdietrich** [comment]
in eclipse the tests are green
ill see what happens in gradle

**Comment by cdietrich** [comment]
gradle tests are green locally as well

**Comment by cdietrich** [comment]
@dhuebner can you try this on your local machine?

**Comment by dhuebner** [comment]
@cdietrich I will check the upstream changes first. xtext-xtend fails too. :(

**Comment by dhuebner** [comment]
@cdietrich Probably caused by the latest xtext-lib change. Thx for verification

**Comment by cdietrich** [comment]
i locally find RequestManagerTest and EncodingTest and UnicodeTest as flaky

**Pull request merged**